### PR TITLE
rd/aws_codebuild_fleet: new resource and data source

### PR DESCRIPTION
**Pull request by kamilturek**
### Description
This PR introduces `aws_codebuild_fleet` resource and data source.

### Relations
Closes https://github.com/hashicorp/terraform-provider-aws/issues/36010.
Extends https://github.com/hashicorp/terraform-provider-aws/pull/36790 (credits to @alexknez).

### References
https://docs.aws.amazon.com/codebuild/latest/APIReference/API_CreateFleet.html

### Output from Acceptance Testing

```console
% make testacc PKG=codebuild ACCTEST_PARALLELISM=1 TESTARGS="-run=TestAccCodeBuildFleet_"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/codebuild/... -v -count 1 -parallel 1  -run=TestAccCodeBuildFleet_ -timeout 360m
=== RUN   TestAccCodeBuildFleet_basic
=== PAUSE TestAccCodeBuildFleet_basic
=== RUN   TestAccCodeBuildFleet_disappears
=== PAUSE TestAccCodeBuildFleet_disappears
=== RUN   TestAccCodeBuildFleet_tags
=== PAUSE TestAccCodeBuildFleet_tags
=== RUN   TestAccCodeBuildFleet_baseCapacity
=== PAUSE TestAccCodeBuildFleet_baseCapacity
=== RUN   TestAccCodeBuildFleet_computeType
=== PAUSE TestAccCodeBuildFleet_computeType
=== RUN   TestAccCodeBuildFleet_environmentType
=== PAUSE TestAccCodeBuildFleet_environmentType
=== RUN   TestAccCodeBuildFleet_imageId
=== PAUSE TestAccCodeBuildFleet_imageId
=== RUN   TestAccCodeBuildFleet_scalingConfiguration
=== PAUSE TestAccCodeBuildFleet_scalingConfiguration
=== RUN   TestAccCodeBuildFleet_vpcConfig
=== PAUSE TestAccCodeBuildFleet_vpcConfig
=== CONT  TestAccCodeBuildFleet_basic
--- PASS: TestAccCodeBuildFleet_basic (42.82s)
=== CONT  TestAccCodeBuildFleet_environmentType
--- PASS: TestAccCodeBuildFleet_environmentType (77.17s)
=== CONT  TestAccCodeBuildFleet_vpcConfig
--- PASS: TestAccCodeBuildFleet_vpcConfig (151.76s)
=== CONT  TestAccCodeBuildFleet_scalingConfiguration
--- PASS: TestAccCodeBuildFleet_scalingConfiguration (77.59s)
=== CONT  TestAccCodeBuildFleet_imageId
--- PASS: TestAccCodeBuildFleet_imageId (42.10s)
=== CONT  TestAccCodeBuildFleet_baseCapacity
--- PASS: TestAccCodeBuildFleet_baseCapacity (77.39s)
=== CONT  TestAccCodeBuildFleet_computeType
--- PASS: TestAccCodeBuildFleet_computeType (77.45s)
=== CONT  TestAccCodeBuildFleet_tags
--- PASS: TestAccCodeBuildFleet_tags (114.51s)
=== CONT  TestAccCodeBuildFleet_disappears
--- PASS: TestAccCodeBuildFleet_disappears (40.25s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/codebuild	705.701s
```

```console
% make testacc PKG=codebuild ACCTEST_PARALLELISM=1 TESTARGS="-run=TestAccCodeBuildFleetDataSource_"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.0 test ./internal/service/codebuild/... -v -count 1 -parallel 1  -run=TestAccCodeBuildFleetDataSource_ -timeout 360m
=== RUN   TestAccCodeBuildFleetDataSource_basic
=== PAUSE TestAccCodeBuildFleetDataSource_basic
=== CONT  TestAccCodeBuildFleetDataSource_basic
--- PASS: TestAccCodeBuildFleetDataSource_basic (40.65s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/codebuild	45.359s
```

### CODE REVIEW

File: internal/service/codebuild/fleet.go [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+	if errs.IsA[*types.ResourceNotFoundException](err) {
+		return diags
+	}
+	if err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionDeleting, ResNameFleet, d.Id(), err)
+	}
+
+	if err := waitFleetDeleted(ctx, conn, d.Id()); err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionWaitingForDeletion, ResNameFleet, d.Id(), err)
+	}
```
Comment (kamilturek) [inline internal/service/codebuild/fleet.go:390]: Fleets can stay in the "pending deletion" state even for a few hours. Not sure what would be the best way to handle such cases (e.g. allow optionally skipping the wait?).

Comment (ewbankkit) [reply internal/service/codebuild/fleet.go:390]: Can fleets be "undeleted"? I not we may just want to treat "pending deletion" as deleted

Comment (kamilturek) [reply internal/service/codebuild/fleet.go:390]: Apparently it cannot. When I try to perform any action on a fleet "pending deletion", I get the following error:
```
Fleet status is PENDING_DELETION_SUBMITTED. Cannot be updated.
```

Therefore, I'll remove the waiter.

Comment (kamilturek) [reply internal/service/codebuild/fleet.go]: @ewbankkit Done ✅

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccCodeBuildFleet_' PKG=codebuild ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/codebuild/... -v -count 1 -parallel 3  -run=TestAccCodeBuildFleet_ -timeout 360m
=== RUN   TestAccCodeBuildFleet_basic
=== PAUSE TestAccCodeBuildFleet_basic
=== RUN   TestAccCodeBuildFleet_disappears
=== PAUSE TestAccCodeBuildFleet_disappears
=== RUN   TestAccCodeBuildFleet_tags
=== PAUSE TestAccCodeBuildFleet_tags
=== RUN   TestAccCodeBuildFleet_baseCapacity
=== PAUSE TestAccCodeBuildFleet_baseCapacity
=== RUN   TestAccCodeBuildFleet_computeType
=== PAUSE TestAccCodeBuildFleet_computeType
=== RUN   TestAccCodeBuildFleet_environmentType
=== PAUSE TestAccCodeBuildFleet_environmentType
=== RUN   TestAccCodeBuildFleet_imageId
=== PAUSE TestAccCodeBuildFleet_imageId
=== RUN   TestAccCodeBuildFleet_scalingConfiguration
=== PAUSE TestAccCodeBuildFleet_scalingConfiguration
=== RUN   TestAccCodeBuildFleet_vpcConfig
=== PAUSE TestAccCodeBuildFleet_vpcConfig
=== CONT  TestAccCodeBuildFleet_basic
=== CONT  TestAccCodeBuildFleet_environmentType
=== CONT  TestAccCodeBuildFleet_baseCapacity
--- PASS: TestAccCodeBuildFleet_basic (44.74s)
=== CONT  TestAccCodeBuildFleet_tags
--- PASS: TestAccCodeBuildFleet_baseCapacity (65.94s)
=== CONT  TestAccCodeBuildFleet_scalingConfiguration
--- PASS: TestAccCodeBuildFleet_environmentType (66.53s)
=== CONT  TestAccCodeBuildFleet_vpcConfig
--- PASS: TestAccCodeBuildFleet_scalingConfiguration (64.53s)
=== CONT  TestAccCodeBuildFleet_imageId
--- PASS: TestAccCodeBuildFleet_tags (87.98s)
=== CONT  TestAccCodeBuildFleet_disappears
--- PASS: TestAccCodeBuildFleet_imageId (43.81s)
=== CONT  TestAccCodeBuildFleet_computeType
--- PASS: TestAccCodeBuildFleet_disappears (57.04s)
--- PASS: TestAccCodeBuildFleet_computeType (65.17s)
--- PASS: TestAccCodeBuildFleet_vpcConfig (229.07s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/codebuild	300.692s
```
```console
% make testacc TESTARGS='-run=TestAccCodeBuildFleetDataSource_' PKG=codebuild ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/codebuild/... -v -count 1 -parallel 3  -run=TestAccCodeBuildFleetDataSource_ -timeout 360m
=== RUN   TestAccCodeBuildFleetDataSource_basic
=== PAUSE TestAccCodeBuildFleetDataSource_basic
=== CONT  TestAccCodeBuildFleetDataSource_basic
--- PASS: TestAccCodeBuildFleetDataSource_basic (42.14s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/codebuild	46.938s
```

**Comment by ewbankkit** [comment]
@kamilturek Thanks for the contribution :tada: :clap:.

**Pull request merged**